lib/daru/io/io.rb

```diff
@@ -124,20 +124,24 @@ def dataframe_write_csv dataframe, path, opts={}
 
       def from_sql dbh, query
         require 'dbi'
+
         sth     = dbh.execute(query)
         vectors = {}
-        fields  = []
-        sth.column_info.each do |c|
-          vectors[c[:name]] = Daru::Vector.new([])
-          vectors[c[:name]].rename c[:name]
-          fields.push(c[:name].to_sym)
+        fields = []
+        sth.column_names.each do |column_name|
+          name = column_name.to_sym
+          vectors[name] = Daru::Vector.new([])
+          vectors[name].rename name
+          fields.push name
         end
-        ds=Daru::DataFrame.new(vectors,order: fields)
+
+        df = Daru::DataFrame.new(vectors, order: fields)
+
         sth.fetch do |row|
-          ds.add_row(row.to_a)
+          df.add_row(row.to_a)
         end
-        ds.update
-        ds
+
+        df
```

Comment:
I added a df.update. I didn't have any strong opinions about calling update, just wasn't sure if it was necessary.